llvm/lib/Transforms/Yk/BlockDisambiguate.cpp

```diff
@@ -0,0 +1,204 @@
+//===- BlockDisambiguate.cpp - Unambiguous block mapping for yk ----===//
+//
+// This pass ensures that yk is able to unambiguously map machine blocks back
```

Comment:
This comment is very long (which is good!) but reads like a murder mystery (which is bad). Let's give the reader an intuition up-front about what they'll see. Maybe something like "LLVM IR blocks that branch to themselves lead to multiple machine-level basic blocks that we cannot unambiguously map back to LLVM IR. This pass splits such blocks into two parts such that the resulting machine-level basic blocks do have an unambiguous mapping back to the IR."

Something I also haven't quite got from the comment is the precise preconditions for when this pass will decide to rewrite things. It's probably worth clarifying that somewhere (not necesarilly up-front though).

Reply:
Can I squash those last three commits to make the review easier for you?

Reply:
Go ahead.